tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/CustomizedCodeUpdateTool/TspClientUpdateTool.cs

```diff
@@ -0,0 +1,639 @@
+                    var newContent = content.Replace("int bar", "string bar");
+                    File.WriteAllText(patch.File, newContent);
+                    applied++;
+                }
+            }
+            session.Status = dryRun ? "AppliedDryRun" : "Applied";
+            SaveSessionToDisk(session);
+            return Task.FromResult(new TspClientUpdateResponse { Session = session, Message = dryRun ? "Dry-run apply complete" : $"Applied {applied} patches" });
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Apply failed: {sessionId}", sessionId);
+            SetFailure();
+            return Task.FromResult(new TspClientUpdateResponse { ResponseError = ex.Message });
+        }
+    }
+
+    private Task<TspClientUpdateResponse> DoStatus(string sessionId, CancellationToken ct)
+    {
+        try
+        {
+            ct.ThrowIfCancellationRequested();
+            var session = RequireSession(sessionId);
+            return Task.FromResult(new TspClientUpdateResponse { Session = session, Message = "OK" });
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Status failed: {sessionId}", sessionId);
+            SetFailure();
+            return Task.FromResult(new TspClientUpdateResponse { ResponseError = ex.Message });
+        }
+    }
+
+    // --------------- CLI Handlers ---------------
+    private async Task HandleRegenerate(InvocationContext ctx, CancellationToken ct)
+    {
+        try
+        {
+            var spec = ctx.ParseResult.GetValueForArgument(_specPathArg);
+            var sid = ctx.ParseResult.GetValueForOption(_sessionIdOpt);
+            var newGen = ctx.ParseResult.GetValueForOption(_generatedNewOpt);
+            var resp = await DoRegenerate(spec, string.IsNullOrWhiteSpace(sid) ? null : sid, string.IsNullOrWhiteSpace(newGen) ? null : newGen, ct);
+            ctx.ExitCode = ExitCode; output.Output(resp);
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "CLI regenerate failed");
+            SetFailure();
+            ctx.ExitCode = ExitCode; output.Output(new TspClientUpdateResponse { ResponseError = ex.Message });
+        }
+    }
+
+    private async Task HandleDiff(InvocationContext ctx, CancellationToken ct)
+    {
+        try
+        {
+            var sid = ctx.ParseResult.GetValueForOption(_sessionIdOpt);
+            var oldGen = ctx.ParseResult.GetValueForOption(_generatedOldOpt);
+            var newGen = ctx.ParseResult.GetValueForOption(_generatedNewOpt);
+            var resp = await DoDiff(RequireSessionIdOrError(sid, ctx), oldGen, newGen, ct);
+            ctx.ExitCode = ExitCode; output.Output(resp);
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "CLI diff failed");
+            SetFailure();
+            ctx.ExitCode = ExitCode; output.Output(new TspClientUpdateResponse { ResponseError = ex.Message });
```

Comment:
```suggestion
            ctx.ExitCode = ExitCode; output.OutputError(new TspClientUpdateResponse { ResponseError = ex.Message });
```

We can output all these failure responses to stderr this way.